test/e2e/page-objects/flows/user-operations.flow.ts

```diff
@@ -0,0 +1,189 @@
+ *
+ * @param driver - The WebDriver instance
+ * @param snapUrl - The URL of the ERC-4337 snap dapp
+ */
+export async function installERC4337Snap(
+  driver: Driver,
+  snapUrl: string,
+): Promise<void> {
+  console.log('Installing ERC-4337 Account Abstraction Snap');
+
+  const erc4337SnapPage = new ERC4337SnapPage(driver);
+  const snapInstall = new SnapInstall(driver);
+
+  // Open snap page and connect
+  await erc4337SnapPage.openSnapPage(snapUrl);
+  await erc4337SnapPage.connectToSnap();
+
+  // Handle snap installation dialog
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+  await snapInstall.clickConnectButton();
+  await snapInstall.checkPageIsLoaded();
+  await snapInstall.clickConfirmButton();
+  await snapInstall.clickOkButton();
+}
+
+/**
+ * Flow for setting up the ERC-4337 snap with configuration and creating an account.
+ *
+ * @param driver - The WebDriver instance
+ * @param config - Configuration object for the snap
+ * @param config.bundlerUrl - The bundler URL
+ * @param config.entrypoint - The entry point address
+ * @param config.simpleAccountFactory - The simple account factory address
+ * @param config.paymaster - Optional paymaster address
+ * @param config.paymasterSK - Optional paymaster secret key
+ * @param accountConfig - Account creation configuration
+ * @param accountConfig.privateKey - The private key for the account
+ * @param accountConfig.salt - The salt for account creation
+ */
+export async function setupERC4337SnapAccount(
+  driver: Driver,
+  config: {
+    bundlerUrl: string;
+    entrypoint: string;
+    simpleAccountFactory: string;
+    paymaster?: string;
+    paymasterSK?: string;
+  },
+  accountConfig: {
+    privateKey: string;
+    salt: string;
+  },
+): Promise<void> {
+  console.log('Setting up ERC-4337 snap account');
+
+  const erc4337SnapPage = new ERC4337SnapPage(driver);
+
+  // Configure the snap
+  await erc4337SnapPage.setSnapConfiguration(config);
+
+  // Create the snap account
+  await erc4337SnapPage.createSnapAccount(
+    accountConfig.privateKey,
+    accountConfig.salt,
+  );
+}
+
+/**
+ * Flow for connecting a test dapp to the ERC-4337 snap account.
+ *
+ * @param driver - The WebDriver instance
+ * @param accountAddress - The public address of the ERC-4337 account
+ */
+export async function connectTestDappToSnapAccount(
+  driver: Driver,
+  accountAddress: string,
+): Promise<void> {
+  console.log('Connecting test dapp to snap account');
+
+  const testDapp = new TestDapp(driver);
+  await testDapp.openTestDappPage();
+  await testDapp.connectAccount({ publicAddress: accountAddress });
+}
+
+/**
+ * Flow for confirming a transaction in the notification window.
+ *
+ * @param driver - The WebDriver instance
+ */
+export async function confirmTransaction(driver: Driver): Promise<void> {
+  console.log('Confirming transaction');
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+  await driver.clickElement({ text: 'Confirm' });
+}
+
```

Comment:
ℹ️  this could be removed from here as it's not related to user-ops.
If we want to use this steps in the user op flow, we could add this inside the main function, and use the page class for confirmation instead of the Confirm selector